### PR TITLE
[Magiclysm] Update proficiency families and weakpoints for Magiclysm monsters

### LINKED ISSUE #70307: Zombified Yule Cat corpse provides no dissection education
### Describe the bug

If you go to dissect a Zombified Yule Cat, you won't learn anything in the process.

### Attach save file

N/A

### Steps to reproduce

Kill a Zombified Yule Cat, try to dissect.

### Expected behavior

I'd expect to learn at least the basics.

### Screenshots

image

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3693 (22H2)
- Game Version: 04b6eb2 [64-bit]
- Graphics Version: Tiles
- Game Language: System language []
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Portal Storms Ignore NPCs [personal_portal_storms],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Tamable Wildlife [Tamable_Wildlife],
    Bionic Professions [package_bionic_professions],
    Magiclysm [magiclysm],
    Blaze Industries [blazeindustries],
    Mind Over Matter [mindovermatter],
    Xedra Evolved [xedra_evolved],
    Translate Complex Dialogue [translate_dialogue],
    Alternative Map Key [alt_map_key],
    Bombastic Perks [bombastic_perks],
    Extra Mutated Scenarios [extra_mut_scens]
]

### Additional context

_No response_

### LINKED ISSUE #77361: [General bug] Can't learn dissecting proficiencies from magiclism monsters and human's bodies that died before game start.
### Is your feature request related to a problem? Please describe.

As the title says, trying to dissect a corpse found in the field, as well as the bodies of monsters from the Magiclysm, will not teach you anything. The latter is especially unpleasant, because i feel that since the corpse does not teach anything, then the advantages associated with proficiency will not work when you fight wih said corpse when it was't corpse yet.

### Solution you would like.

Proficiencies should be learnable and the benefits of proficiencies must be applied when you fight.

### Describe alternatives you have considered.

none

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jpfalk1**
#### Summary
Mods "Reviewed Magiclysm Monsters and added proficiency families and weakpoints when missing."

#### Purpose of change
Two related issues noted missing proficiencies and weakpoints in Magiclysm monsters. Resolves #70307, resolves #77361. Black Dragons deserve a custom set of weakpoints, so will be addressed with a separate PR.

#### Describe the solution
Added existing applicable proficiency families (mostly "prof_gross_anatomy", "prof_intro_biology", and "prof_physiology") and weakpoints (mostly "wps_humanoid_body", and "wps_animal_quadruped") to Magiclysm monsters. Using existing proficiency and weakpoint sets from similar base CDDA monsters (i.e. Owlbear has similar proficiencies and weakpoints as a Bear, since it doesn't fly)
Demon Spiders also received applicable bug/spider proficiencies and weakpoints. Some of their general data was also moved to the mod-wide file that was appropriate (i.e. Demon Spider species data moved to data/mods/Magiclysm/species.json). An abstract monster type for demon spiders was also created, although this may have been unnecessary complexity.
Demihuman monsters were updated to consistently include "prof_gross_anatomy" and "prof_intro_biology". Most already had weakpoints defined and the "demihuman" proficiency.
Black Dragons deserve their own unique weakpoints and proficiencies so will be addressed with a separate PR as this one only used existing proficiencies and weakpoints.
Spirits, holograms, summons, and other related monsters that don't leave a corpse or were primarily intended as player summons were left undefined.
Golems were left undefined for proficiencies / weakpoints, consistent with robots in base CDDA.

#### Describe alternatives you've considered
Considered a new set of proficiencies for all Magiclysm monsters (i.e. "Magical Creature") but this seemed overkill for the majority of the creatures so existing proficiencies were used instead (again, except Black Dragons).
Considered leaving undefined, treating Magiclysm monsters are "too different" to have weakpoints. A lot of the demi-human monsters already had proficiencies/weakpoints defined, it was the odd creatures like Yule Cats and such that did not.

#### Testing
Created a local mod and spawned a sampling of affected monsters to confirm dissection provided the relevant proficiency training. Did not have a good consistent test to assess weakpoints, or appropriateness of selected weakpoints.

### CODE REVIEW

**Comment by jpfalk1** [comment]
> Spell checker encountered unrecognized words in the in-game text added in this pull request. See below for details.

Fixed.

In general did not do as good a job as I thought in setting up the editor to match the JSON style guide. Will be sure to use the linter in the future.

**Comment by jpfalk1** [comment]
Basic build didn't like the "." in the abstract description. Removed.

File: data/mods/Magiclysm/monsters/demon_spider.json [hunk 1]
```diff
@@ -1,43 +1,31 @@
...
-    ]
-  },
-  {
-    "type": "MONSTER_FACTION",
-    "name": "demon_spider",
-    "base_faction": "spider"
+    "abstract": "demon_spider_base",
+    "type": "MONSTER",
+    "name": "base demon spider",
+    "description": "abstract demon spider you shouldn't see this",
```
Comment (Night-Pryanik) [inline data/mods/Magiclysm/monsters/demon_spider.json:6]:
```suggestion
    "description": { "str": "abstract demon spider you shouldn't see this", "//~": "NO_I18N" },
```

File: data/mods/Magiclysm/monsters/monsters.json [hunk 2]
```diff
@@ -564,6 +585,19 @@
...
+        "id": "wing",
+        "name": "the wings",
+        "crit_mult": { "all": 0.75 },
+        "difficulty": { "ranged": 2, "melee": 1 },
+        "coverage_mult": { "point": 0.25 },
+        "effects": [ { "effect": "staggered", "chance": 15, "message": "The %s is knocked off-balance!", "damage_required": [ 10, 100 ] } ],
+        "coverage": 12
+      }
+    ],
+    "families": [ "prof_gross_anatomy", "prof_intro_biology", "prof_wp_demihuman" ],
```
Comment (Maleclypse) [inline data/mods/Magiclysm/monsters/monsters.json:600]:
```suggestion
```
Demons of shadow and fire aren't mortal and  shouldn't have mortal weakpoints.  At most they should have unique weakpoint sets like Dragons.

File: data/mods/Magiclysm/monsters/monsters.json [hunk 3]
```diff
@@ -872,6 +906,8 @@
     "vision_night": 15,
     "harvest": "zombie_animal",
     "//": "Not a zombie, but otyughs are so filthy and diseased their meat is tainted anyway.",
+    "weakpoint_sets": [ "wps_animal_quadruped" ],
+    "families": [ "prof_gross_anatomy", "prof_intro_biology", "prof_physiology", "prof_wp_ungulate" ],
```
Comment (Maleclypse) [inline data/mods/Magiclysm/monsters/monsters.json:910]:
```suggestion
```
Too unique for these.

**Review by Maleclypse — CHANGES_REQUESTED** [review]

File: data/mods/Magiclysm/monsters/monsters.json [hunk 4]
```diff
@@ -531,7 +552,7 @@
     "dodge": 6,
     "vision_night": 5,
     "harvest": "demihuman_fur",
-    "families": [ "prof_wp_demihuman" ],
+    "families": [ "prof_gross_anatomy", "prof_intro_biology", "prof_wp_demihuman" ],
```
[the remark this answers is not in the document]
Comment (jpfalk1) [reply data/mods/Magiclysm/monsters/monsters.json]: Good suggestion, changed the families as suggested and also changed the weakpoint_set to "quadruped".

**Pull request merged**